saltgui/static/scripts/api.js

```diff
@@ -74,6 +74,14 @@ class API {
     if(isShowing && evt.target.className !== "popup") return;
     manualRun.style.display = isShowing ? "none" : "block";
     document.body.style["overflow-y"] = isShowing ? "scroll" : "hidden";
+
+    // test whether the command may have caused an update to the list
+    // the user may have altered the text after running the command, just ignore that
+    var command = document.querySelector(".run-command #command").value;
+    var output = document.querySelector(".run-command pre").innerHTML;
+    if(isShowing && command.startsWith("salt.wheel.key.") && output != "Waiting for command...") {
+      location.reload(); 
+    }
```

Comment:
Hmmm, I'm not really happy with this solution. Wouldn't it be better to store something in a state which decides to reload the page or not?

Reply:
The user has the freedom to change commands. That is why the actual command that is used needs to be tested. I do intend to remove this crude function and replace it with a mechanism based on the information from either "GET /events" and/or "GET /ws", In that case the server is pushing changes back to the client. The result is a dynamically updating screen. Also when actions are performed outside SaltGUI. My proposal is to leave this code until that is done.

Reply:
Ok, then we will leave it for now.